op-e2e/migration_test.go

```diff
@@ -0,0 +1,376 @@
+package op_e2e
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/ethereum-optimism/optimism/op-bindings/hardhat"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis/migration_action"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils"
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum-optimism/optimism/op-node/metrics"
+	"github.com/ethereum-optimism/optimism/op-node/node"
+	"github.com/ethereum-optimism/optimism/op-node/p2p"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/driver"
+	"github.com/ethereum-optimism/optimism/op-node/testlog"
+	"github.com/ethereum-optimism/optimism/op-service/backoff"
+	"github.com/ethereum/go-ethereum/eth/ethconfig"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+	"time"
+)
+
+type migrationTestConfig struct {
+	enabled           bool
+	l1URL             string
+	l2Path            string
+	ovmAddrsPath      string
+	evmAddrsPath      string
+	ovmAllowancesPath string
+	ovmMessagesPath   string
+	evmMessagesPath   string
+}
+
+var config migrationTestConfig
+
+var cwd string
+
+func init() {
+	if os.Getenv("OP_E2E_MIGRATION_ENABLED") != "true" {
+		return
+	}
+
+	iCwd, err := os.Getwd()
+	if err != nil {
+		panic("failed to get cwd")
+	}
+	cwd = iCwd
+
+	config.enabled = true
+	config.l1URL = os.Getenv("OP_E2E_MIGRATION_L1_URL")
+	if config.l1URL == "" {
+		panic("must specify an L1 url")
+	}
+	config.l2Path = os.Getenv("OP_E2E_MIGRATION_L2_DATA_PATH")
+	if config.l2Path == "" {
+		panic("must specify an l2 data path")
+	}
+
+	migrationDataDir := path.Join(cwd, "..", "packages", "migration-data", "data")
+	config.ovmAddrsPath = path.Join(migrationDataDir, "ovm-addresses.json")
+	config.evmAddrsPath = path.Join(migrationDataDir, "evm-addresses.json")
+	config.ovmAllowancesPath = path.Join(migrationDataDir, "ovm-allowances.json")
+	config.ovmMessagesPath = path.Join(migrationDataDir, "ovm-messages.json")
+	config.evmMessagesPath = path.Join(migrationDataDir, "evm-messages.json")
+}
+
+type storageSlot struct {
+	addr  string
+	slot  string
+	value string
+}
+
+const (
+	networkName  = "mainnet-forked"
+	hardhatImage = "docker.io/ethereumoptimism/hardhat-node:latest"
+)
+
```

Comment:
Having some comments here for what each storage slot references/does would be great